snaps/how-to/use-environment-variables.md

```diff
@@ -0,0 +1,90 @@
+---
+sidebar_position: 3
+---
+
+# Use environment variables
+
+You can use environment variables to compile your Snap.
+This is useful if you want to use different variables for testing your Snap locally and publishing
+your Snap.
+You can use environment variables [on the command line](#use-environment-variables-on-the-command-line)
+or [in a `.env` file](#use-environment-variables-in-a-env-file).
+
+## Use environment variables on the command line
+
+1. Specify environment variables on the command line.
+    For example:
+
+    ```bash
+    PUBLIC_KEY=abc123 SNAP_ENV=dev
+    ```
+
+2. Build your Snap using the [Snaps CLI](../reference/cli/subcommands.md):
+
+    ```bash
+    yarn mm-snap build
+    ```
+
+## Use environment variables in a `.env` file
+
+1. Specify environment variables in a `.env` file.
+    For example:
+    
+    ```text title=".env"
+    PUBLIC_KEY=abc123
+    SNAP_ENV=dev
+    ```
+
+2. Install the [`envify`](https://github.com/hughsk/envify) package using Yarn or npm:
+
+    ```bash
+    yarn add envify
+    ```
+    
+    or
+    
+    ```bash
+    npm install envify
+    ```
+
+3. Use the environment variables in your Snap's
+    [configuration file](../learn/about-snaps/files.md#configuration-file) by modifying the bundler object.
+    For example:
+
+    ```javascript title="snap.config.js"
+    const envify = require("envify/custom");
+    require("dotenv").config();
+    
+    module.exports = {
+        cliOptions: {
+            src: "./src/index.ts",
+            port: 8080,
+        },
+        bundlerCustomizer: (bundler) => {
+            bundler.transform(
+                envify({
+                    SNAP_ENV: process.env.SNAP_ENV,
+                    PUBLIC_KEY: process.env.PUBLIC_KEY,
+                }),
+            );
+        },
+    };
```

Comment:
This is using the deprecated Browserify config. We should recommend the Webpack based config, which has better support for environment variables: https://github.com/MetaMask/snaps/tree/main/packages/snaps-cli#environment

Reply:
@Mrtenz Can you check that I've updated the environment config correctly? I also noticed that our Snaps CLI docs are quite outdated, raised a separate issue here: #1169.